aws_lambda_powertools/utilities/data_classes/alb_event.py

```diff
@@ -37,25 +38,15 @@ def multi_value_query_string_parameters(self) -> Dict[str, List[str]]:
 
     @property
     def resolved_query_string_parameters(self) -> Dict[str, List[str]]:
-        if self.multi_value_query_string_parameters:
-            return self.multi_value_query_string_parameters
-
-        return super().resolved_query_string_parameters
+        return self.multi_value_query_string_parameters or super().resolved_query_string_parameters
 
     @property
-    def resolved_headers_field(self) -> Dict[str, Any]:
-        headers: Dict[str, Any] = {}
-
-        if self.multi_value_headers:
-            headers = self.multi_value_headers
-        else:
-            headers = self.headers
-
-        return {key.lower(): value for key, value in headers.items()}
+    def multi_value_headers(self) -> Dict[str, List[str]]:
+        return CaseInsensitiveDict(self.get("multiValueHeaders"))
```

Comment:
Yes. The CaseInsensitiveDict is basically a Dict that transforms the key to lower in all operations. The side-effect is that the keys are case-insensitive. E.g. doing `header["Authorization"] = "Bearer x"` or `header.get("Authorization")` will result the same as `header["Authorization".lower()] = "Bearer x"` or `header.get("Authorization".lower())` because of how keys are transformed internally.